tests/integ/test_sagemaker_steps.py

```diff
@@ -104,6 +105,65 @@ def test_training_step(pca_estimator_fixture, record_set_fixture, sfn_client, sf
         state_machine_delete_wait(sfn_client, workflow.state_machine_arn)
         # End of Cleanup
 
+
+def test_training_step_with_placeholders(pca_estimator_fixture,
+                                         record_set_fixture,
+                                         sfn_client,
+                                         sfn_role_arn,
+                                         sagemaker_session):
+    # Build workflow definition
+    execution_input = ExecutionInput(schema={
+        'JobName': str,
+        'OutputPath': str,
+        'Data': str
+    })
+
+    job_name_placeholder = execution_input['JobName']
+    output_path_placeholder = execution_input['OutputPath']
+    training_step_name = 'TrainingStep'
+
+    training_step = TrainingStep(training_step_name,
+                                 estimator=pca_estimator_fixture,
+                                 job_name=job_name_placeholder,
+                                 data=record_set_fixture,
+                                 mini_batch_size=200,
+                                 output_path=output_path_placeholder)
+    workflow_graph = Chain([training_step])
+
+    with timeout(minutes=DEFAULT_TIMEOUT_MINUTES):
+        # Create workflow and check definition
+        workflow = create_workflow_and_check_definition(
+            workflow_graph=workflow_graph,
+            workflow_name=unique_name_from_base("integ-test-training-step-workflow"),
+            sfn_client=sfn_client,
+            sfn_role_arn=sfn_role_arn
+        )
+
+        # Execute workflow
+        job_id = generate_job_name()
+        execution_input = {
+            'OutputPath': f's3://{sagemaker_session.default_bucket()}/',
+            'JobName': f'TrainingJob-{job_id}'
+        }
+
+        # Check placeholders in workflow definition:
+        execution = workflow.execute(inputs=execution_input)
+        expected_output_data_key_value_pair = ("S3OutputPath.$", "$$.Execution.Input['OutputPath']")
+        assert expected_output_data_key_value_pair in \
+               workflow.definition.states[training_step_name]['Parameters']['OutputDataConfig'].items()
+
+        expected_job_name_key_value_pair = ("TrainingJobName.$", "$$.Execution.Input['JobName']")
+        assert expected_job_name_key_value_pair in \
+               workflow.definition.states[training_step_name]['Parameters'].items()
```

Comment:
Static ASL validation is already covered in the unit tests and doesn't require an execution. `create_workflow_and_check_definition` already asserts that the created definition from DescribeStateMachine matches the python object.

Reply:
You are right - will remove the asserts!